apps/trip_plan/lib/trip_plan/api/open_trip_planner.ex

```diff
@@ -26,12 +26,54 @@ defmodule TripPlan.Api.OpenTripPlanner do
           }
         )
 
-      root_url = params["root_url"] || config(:root_url)
+      root_url = params["root_url"] || pick_url(connection_opts)
       full_url = "#{root_url}/otp/routers/default/plan"
       send_request(full_url, params, &parse_json/1)
     end
   end
 
+  def pick_url(connection_opts) do
+    cond do
+      connection_opts[:force_otp2] ->
+        config(:otp2_url)
+
+      connection_opts[:force_otp1] ->
+        config(:otp1_url)
+
+      true ->
+        user_id = connection_opts[:user_id]
+        percent_threshold = get_otp2_percentage()
+
+        :rand.seed(:exsss, user_id)
+        placement = :rand.uniform()
+        use_otp2 = placement < percent_threshold / 100
+
+        Logger.info(fn ->
+          "#{__MODULE__}.pick_url placement=#{placement} otp2_percentage=#{percent_threshold}% mbta_id=#{user_id} use_otp2=#{use_otp2}"
```

Comment:
I thought about it, but wasn't sure if it would be worth it, since it would only be used internally, but I'd be happy to, it might be useful if we're trying to test one or the other to make sure we're hitting what we expect.